node/node.go

```diff
@@ -129,7 +129,8 @@ func New(cfg *Config, version string) (*Node, error) { //nolint:gocyclo,funlen
 
 	services := make([]service.Service, 0)
 
-	chain := blockchain.New(database, &cfg.Network)
+	synchronizer := new(sync.Synchronizer)
```

Comment:
Looks like we create synchronizer and they depend on each other. What were other alternatives to puting pending block in synchronizer? (not going to ask you to rewrite it, I understand that this written like this for a reason)

Reply:
Yeah, this was something I thought about too. I don't think the dependency between the synchroniser and the pending can be removed since the synchroniser updates the pending block and state. One potential solution is to have pending in its own package and let synchroniser import it but I think this will be just a cosmetic change (i.e. nothing meaningful).